pkg/util/deployment/deployment.go

```diff
@@ -270,6 +279,16 @@ func waitForReplicaSetUpdated(c clientset.Interface, desiredGeneration int64, na
 	})
 }
 
+func waitForPodsHashPopulated(c clientset.Interface, namespace, name string) error {
```

Comment:
Actually, better in pkg/client/unversioned/conditions.go

WaitForObservedDeployment probably belongs there, also, instead of in pkg/util/deployment/deployment.go